File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/DicerDropsJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class DicerDropsJson {
    @Expose
    public DicerType MELON;

    @Expose
    public DicerType PUMPKIN;

    public static class DicerType {
        @Expose
        @SerializedName("total chance")
        public Integer totalChance;

        @Expose
        public List<DropInfo> drops;
    }

    public static class DropInfo {
        @Expose
        public Integer chance;

        @Expose
        public List<Integer> amount;
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/DicerDropsJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class DicerDropsJson(
    @Expose val MELON: DicerType,
    @Expose val PUMPKIN: DicerType
)

data class DicerType(
    @Expose @SerializedName("total chance") val totalChance: Int,
    @Expose val drops: List<DropInfo>
)

data class DropInfo(
    @Expose val chance: Int,
    @Expose val amount: List<Int>
)
